tests/test_math.hh

```diff
@@ -481,7 +483,7 @@ namespace rascal {
 
     ~Hyp1f1GradientProvider() = default;
 
-    Eigen::Ref<Eigen::Array<double, 1, Eigen::Dynamic>>
+    Eigen::Array<double, 1, Eigen::Dynamic>
     f(const Eigen::Matrix<double, 1, 1> & input_v) {
       this->hyp1f1_calculator.calc(input_v(0), this->fac_a, this->fac_b);
       Eigen::MatrixXd result(this->max_radial, this->max_angular + 1);
```

Comment:
This and the next line are making a copy of the data stored inside `this->hyp1f1_calculator`, right? The memory error was that the `Eigen::Ref` was referring to memory allocated by `result`, and freed as `result` got out of scope
at the end of this function.

Could we not make this copy, and instead return a `Eigen::Ref<...>(this->hyp1f1_calculator.get_data())` directly?

Reply:
Please, you're making my head hurt. This is not only a bad idea (see previous comment about aliasing issues) but it's also way overkill for the simple fix that this is. For perspective: This is (a) testing code that's (b) run on only one function that (c) only copies a small amount of data for each iteration of the gradient test that (d) takes a small proportion of the testing time anyway. Let's let Eigen and the compiler worry about superfluous copies; we need to worry more about the correctness of our code.

Reply:
yeah, sorry I went a bit into overdrive here, I did not realize this was test only code =)